akka-http-testkit/src/test/scala/akka/http/scaladsl/testkit/ScalatestRouteTestSpec.scala

```diff
@@ -42,6 +43,20 @@ class ScalatestRouteTestSpec extends AnyFreeSpec with Matchers with ScalatestRou
       }
     }
 
+    "a test using ~!> and some checks" in {
+      // raw here, should have been parsed into modelled header when going through an actual server when using `~!>`
+      val extraHeader = RawHeader("X-Forwarded-Proto", "abc")
+      Get() ~> addHeader(extraHeader) ~!> {
+        respondWithHeader(extraHeader) {
```

Comment:
This test might be clearer if we respond with a completely different header?

Reply:
The `addHeader` a line above seemed to be useless so I just removed that.